doc/specs/#4993 - Keyboard Selection/Keyboard-Selection.md

```diff
@@ -110,25 +117,117 @@ For `SelectionExpansion = Buffer`, the selection endpoint will be moved to the b
+
+Copy Mode is a mode where the user can create and modify a selection using only the keyboard. The following flowchart covers how the new `copymode()` action works:
+
+![Copy Mode Flowchart][images/CopyModeFlowchart.png]
+
+**NOTE**: `copyMode()` refers to the action, whereas `updateSelection()` refers to the underlying function that is being called in the code.
+
+If a selection is not active, a "start" and "end" selection point is created at the cursor position. `updateSelection()` calls then move "start" and "end" together as one position.
+
+Invoking `copyMode()` again, will then anchor "start" (meaning that it will be kept in place). Subsequent `updateSelection()` calls move the "end" selection point.
+
+Invoking `copyMode()` essentially cycles between which selection point is targeted.
 
 ## UI/UX Design
 
 ### Key Bindings
 
-There will only be 1 new command that needs to be added:
 | Action | Keybinding Args | Description |
 |--|--|--|
-| `selectAll`  | | Select the entire text buffer.
+| `selectAll`                    | none | Select the entire text buffer. |
+| `markMode`                     | none | Toggle mark mode. If no selection exists, create a selection at the cursor position. Otherwise, use the existing selection as one in mark mode. |
+| `toggleBlockSelection`         | none | Transform the existing selection between a block selection and a line selection.  |
+| `switchSelectionEndpoint`      | none | If a selection is present, switch which selection endpoint is targeted when in mark mode or quick edit mode. |
 
 By default, the following key binding will be set:
 ```JS
 { "command": "selectAll", "keys": "ctrl+shift+a" },
+
+// Copy Mode
+{ "command": "copyMode", "keys": "ctrl+shift+m" },
+{ "command": "toggleBlockSelection" },
+{ "command": "switchSelectionEndpoint" },
 ```
 
+### Y-Beam
+
+A y-beam will be used to identify which selection endpoint is currently being moved when in mark mode.
+
+When we're moving the cursor (this happens when mark mode is entered from no existing selection), a full y-beam will be displayed at the cursor position.
+
+![Y-Beam Example][./images/Y-Beam.png]
+
+When <kbd>shift</kbd> is held, we're expanding the selection. In this case, the y-beam will be split, and the relevant half will be rendered on the active endpoint.
+
+![Separated Y-Beam Example][./images/Half-Y-Beam.png]
+
+**NOTE:** Both half y-beams could have been presented as shown in the image below. This idea was omitted because then there is no indication for which half y-beam is currently focused.
+
+![Both Separated Y-Beams Example][./images/Split-Y-Beam.png]

```

Comment:
Off-handedly had a conversation with Dustin about this. I should update it to include the following...
- How does this work with line wrapping?
	- someday, we'll have (1) selections with inclusive starts and exclusive ends and (2) a text buffer that can keep track of newlines (instead of just filling it with blank spaces). When that happens, we should be able to select before and after the newline character, as opposed to each individual space.
- Should we use a different cursor for block selection?
	- imagine using some kind of L-shaped cursor!
- I should probably mention the selection background and font size changes